src/mame/includes/p2000t.h

```diff
@@ -27,16 +28,20 @@ class p2000t_state : public driver_device
 		, m_maincpu(*this, "maincpu")
 		, m_speaker(*this, "speaker")
 		, m_mdcr(*this, "mdcr")
+		, m_ram(*this, RAM_TAG)
+		, m_bank(*this, "bank")
 		, m_keyboard(*this, "KEY.%u", 0)
 	{
 	}
 
 	void p2000t(machine_config &config);
+	virtual void machine_start() override;
+
 
 protected:
 	uint8_t p2000t_port_000f_r(offs_t offset);
-	uint8_t p2000t_port_202f_r();
-	void p2000t_port_101f_w(uint8_t data);
+	virtual uint8_t p2000t_port_202f_r();
+	virtual void p2000t_port_101f_w(uint8_t data);
```

Comment:
Why do these need to be virtual?  I may be missing something, but I can’t see them overridden anywhere.